gamersnet_backend/routes/posts/getPosts.js

```diff
@@ -1,10 +1,51 @@
 'use strict';
 
 
-// include our function from the database to add post
-let {getAllPosts, getValidPosts} = require('../../persistence/posts');
+// include our function from the database to fetch posts
+let {getUserPosts, getAllPosts, getValidPosts} = require('../../persistence/posts');
 
-// this function handles the /post/getAllPosts/ endpoint
+let {verifyUserLoggedIn} = require('../utilities/tokenUtility')
+
+let {getUserIDFromToken} = require('../../persistence/tokens.js')
+
+// this function handles the /posts/listUserPosts/ endpoint
+async function listUserPosts(request, response) {
```

Comment:
From the looks of it, it appears that this route fetches all posts related to a particular user. However, the route checks if the client is logged in before fetching posts of a user, was this intended?

Reply:
Well, yes! This route will be used for 2 purposes; 

- when any user visits their own profile to see their posts (he has to be logged in)
- when any user visits another user's profile to see that users' posts, they have to be logged in order to see other user's profile

Note: this route will not be used with search, and checking for logged in is only intended if the user wants to see another user/their own profile. May/Should create 2 routes if that's what we need!